static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -11,18 +11,49 @@ import { isBlocked } from 'common/modules/commercial/membership-engagement-banne
 import { getSync as getGeoLocation } from 'lib/geolocation';
 import { shouldShowReaderRevenue } from 'common/modules/commercial/contributions-utilities';
 import type { Banner } from 'common/modules/ui/bannerPicker';
+import bean from 'bean';
+import fetchJson from 'lib/fetch-json';
+import reportError from 'lib/report-error';
 
 import {
     submitComponentEvent,
     addTrackingCodesToUrl,
 } from 'common/modules/commercial/acquisitions-ophan';
 import { acquisitionsBannerControlTemplate } from 'common/modules/commercial/templates/acquisitions-banner-control';
+import userPrefs from 'common/modules/user-prefs';
 
-// change messageCode to force redisplay of the message to users who already closed it.
-const messageCode = 'engagement-banner-2018-07-13';
+type BannerDeployLog = {
+    time: string,
+};
+
+const messageCode = 'engagement-banner';
 
-const canDisplayMembershipEngagementBanner = (): Promise<boolean> =>
-    Promise.resolve(shouldShowReaderRevenue());
+const getTimestampOfLastBannerDeploy = (): Promise<string> =>
+    fetchJson('/reader-revenue/contributions-banner-deploy-log', {
+        mode: 'cors',
+    }).then((resp: BannerDeployLog) => resp && resp.time);
+
+const hasBannerBeenRedeployedSinceClosed = (): Promise<boolean> =>
+    getTimestampOfLastBannerDeploy()
+        .then(timestamp => {
+            const bannerLastDeployedAt = new Date(timestamp);
+            const userLastClosedBannerAt = new Date(
```

Comment:
It might be better to be explicit about the case where `userPrefs.get('engagementBannerLastClosedAt')` is `null`.

```javascript
const engagementBannerLastClosedAt = userPrefs.get('engagementBannerLastClosedAt');
if (!engagementBannerLastClosedAt) {
  // do show the banner if the user's never closed one before
  return true;
}

...
```

Reply:
done!